env.d/60-flatpak

```diff
@@ -0,0 +1,15 @@
+# flatpak-environment-generator
+
+#!/bin/bash
+
+if [ "${XDG_DATA_DIRS#*flatpak}" = "${XDG_DATA_DIRS}" ]; then
+    OLD_XDG_DATA_DIRS=${XDG_DATA_DIRS}
+    XDG_DATA_DIRS="${XDG_DATA_HOME:-"$HOME/.local/share"}/flatpak/exports/share"
+    while read install_path
+    do
+        XDG_DATA_DIRS="${XDG_DATA_DIRS}:$install_path/exports/share"
+    done < <(flatpak --list-installations)
+    XDG_DATA_DIRS="${XDG_DATA_DIRS}:${OLD_XDG_DATA_DIRS:-/usr/local/share:/usr/share}"
+fi
+
+echo XDG_DATA_DIRS=$XDG_DATA_DIRS
```

Comment:
As defensive shell scripting, this should be quoted: `echo "XDG_DATA_DIRS=$XDG_DATA_DIRS"`